File: src/csharp/Intel/Iced/Intel/EncoderException.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if ENCODER
using System;
using System.Runtime.Serialization;

namespace Iced.Intel {
	/// <summary>
	/// Thrown if the encoder can't encode an instruction
	/// </summary>
	[Serializable]
	public class EncoderException : Exception {
		/// <summary>
		/// The instruction that couldn't be encoded
		/// </summary>
		public Instruction Instruction { get; }

		/// <summary>
		/// Constructor
		/// </summary>
		/// <param name="message">Exception message</param>
		/// <param name="instruction">Instruction</param>
		public EncoderException(string message, in Instruction instruction) : base(message) => Instruction = instruction;

		/// <summary>
		/// Constructor
		/// </summary>
		/// <param name="info"></param>
		/// <param name="context"></param>
		protected EncoderException(SerializationInfo info, StreamingContext context) : base(info, context) { }
	}
}
#endif

File: src/java/iced-x86/src/main/java/com/github/icedland/iced/x86/enc/EncoderException.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.enc;

import com.github.icedland.iced.x86.Instruction;

/**
 * Thrown by an {@link Encoder} when it fails to encode an instruction.
 */
public final class EncoderException extends RuntimeException {
	/**
	 * The instruction the {@link Encoder} failed to encode
	 */
	public final Instruction instruction;

	EncoderException(String message, Instruction instruction) {
		super(message);
		this.instruction = instruction;
	}

	private static final long serialVersionUID = 1;
}
